jwst/tweakreg/tweakreg_catalog.py

```diff
@@ -56,6 +58,10 @@ def make_tweakreg_catalog(model, kernel_fwhm, snr_threshold, sharplo=0.2,
     sources = daofind(model.data)
 
     columns = ['id', 'xcentroid', 'ycentroid', 'flux']
-    catalog = sources[columns]
+    if len(sources) > 0:
+        catalog = sources[columns]
+    else:
+        catalog = Table(names=columns, dtype=(np.int_, np.float_, np.float_, np.float_))
+        catalog.add_row([1, 0.5, 0.5, 100.])
```

Comment:
Why are you adding a dummy source here?

Reply:
Why not just return an empty table and check for its length when you need to use it?

Reply:
@larrybradley Because I was too tired to remove debugging code... Thanks for looking at this with fresh eyes :smile: I'll remove it.

Reply:
> Why not just return an empty table and check for its length when you need to use it?

Because saving empty catalogs would produce empty files (see https://github.com/spacetelescope/jwst/blob/d7daa6c9cc60fc9870add2950ca06a424056db28/jwst/tweakreg/tweakreg_step.py#L84). Catalog files with headers but no sources look much better (IMO) and less worrisome than completely empty files, if we even can save empty tables - I get a crash:

```python
ValueError: Inconsistent data column lengths: set()
```

Reply:
Why do you need to save an empty table?  Does your rely on a saved catalog (not in memory)?

Reply:
No, it does not, but the pipeline can save catalogs to files. As an user it would be weird to see that catalogs have been saved only for some images but not for others. Having a file with a header (but no sources) is much more informative (instead of being puzzling).

Reply:
Can't we just issue a warning?

Reply:
We could if that's fine with @hbushouse . However, it would still be puzzling not to find all files and searching through the pipeline log to figure out why some catalogs have not been written out seems like a less obvious solution.

Reply:
For the purpose of running in a pipeline, where tweakreg is just 1 step and other steps could continue to work without it, I'd like to see just a warning issued and the step return with a "skipped" status. The remaining `calwebb_image3` steps, like `skymatch`, `outlier_detection`, and `resample` can still be executed using the original (untweaked) WCS info.

Reply:
In that case I think it makes sense to just issue a warning for the particular catalog that had no sources, and move on with the rest of processing (without skipping the entire step). And let the logic that's already there later on figure out if it really has enough catalog data to work with.

Reply:
@hbushouse That's being done in https://github.com/spacetelescope/jwst/blob/1316c3f6a1ccf767698dca30bbdf9e1a86112ef5/jwst/tweakreg/tweakreg_step.py#L71

Reply:
@mcara For zero detected sources, I would change that from an INFO message to a warning.

Reply:
@larrybradley Done